runtime/src/accounts_db.rs

```diff
@@ -289,13 +289,16 @@ impl AccountStorageEntry {
         if count > 0 {
             *count_and_status = (count - 1, status);
         } else {
-            warn!("count value 0 for slot {}", self.slot_id);
+            panic!("count value 0 for slot {}", self.slot_id);
         }
         count_and_status.0
     }
 
     pub fn set_file<P: AsRef<Path>>(&mut self, path: P) -> IOResult<()> {
-        self.accounts.set_file(path)
+        self.accounts.set_file(path)?;
+        let mut count_and_status = self.count_and_status.write().unwrap();
+        count_and_status.0 = self.accounts.accounts(0).len();
```

Comment:
By always using the count of stored accounts in `AppendVec`, it'll become common for the gabage to pile up quickly between snapshot save/restore. For example, updating accounts started to only increase the count. And I think the count will not be decreased ever. Is that a desired behavior? Is this PR a workaround? Assuming restoring from snapshot is rare. It might be tolerated.

Reply:
I'm not fully sure the state transition of status yet. But, is that safe mutating count other than `AccountStorageStatus::Available`? Or is it ensured that serialized storage entry is always `Available`?

Reply:
Well, restoring is just a one-time practice at validator startup.

Reply:
Anyway, might be good thing than hard error, where we're at the current situation...

Reply:
> Well, restoring is just a one-time practice at validator startup.

@sakridge Yeah. So, I guess this is OK because you're very aware of it. Thanks for confirming!

Reply:
Actually, you are right, this adds too much to the count.

Reply:
@ryoqun as far as the status. I can't think of a case where the status matters. Since it is importing a rooted bank/slot, it would not store any more account updates to the storage files. Even if it does, it would try to append, if it fails then it would update the status to Full.